includes/admin/class.llms.admin.setup.wizard.php

```diff
@@ -25,331 +25,72 @@
-
-		$text = __( 'Skip this step', 'lifterlms' );
-
-		if ( 'coupon' === $step ) {
-			$text = __( 'No thanks', 'lifterlms' );
-		}
-
-		/**
-		 * Filter the skip button text for a given step in the setup wizard
-		 *
-		 * The dynamic portion of this hook, `$step`, refers to the slug of the current step.
-		 *
-		 * @since 4.8.0
-		 *
-		 * @param string $text Button text string.
-		 */
-		return apply_filters( "llms_setup_wizard_get_{$step}_skip_text", $text );
-
-	}
-
-	/**
-	 * Get the URL to a step
-	 *
-	 * @since 3.0.0
-	 *
-	 * @param string $step Step slug.
-	 * @return string
-	 */
-	private function get_step_url( $step ) {
-		return add_query_arg(
-			array(
-				'page' => 'llms-setup',
-				'step' => $step,
+		$this->id    = 'setup';
+		$this->title = __( 'LifterLMS Setup Wizard', 'lifterlms' );
+		$this->steps = array(
+			'intro'    => array(
+				'title' => __( 'Welcome!', 'lifterlms' ),
+				'save'  => __( 'Save & Continue', 'lifterlms' ),
+				'skip'  => __( 'Skip this step', 'lifterlms' ),
+			),
+			'pages'    => array(
+				'title' => __( 'Page Setup', 'lifterlms' ),
+				'save'  => __( 'Save & Continue', 'lifterlms' ),
+				'skip'  => __( 'Skip this step', 'lifterlms' ),
+			),
+			'payments' => array(
+				'title' => __( 'Payments', 'lifterlms' ),
+				'save'  => __( 'Save & Continue', 'lifterlms' ),
+				'skip'  => __( 'Skip this step', 'lifterlms' ),
+			),
+			'coupon'   => array(
+				'title' => __( 'Coupon', 'lifterlms' ),
+				'save'  => __( 'Allow', 'lifterlms' ),
+				'skip'  => __( 'Skip this step', 'lifterlms' ),
+			),
+			'finish'   => array(
+				'title' => __( 'Finish!', 'lifterlms' ),
+				'save'  => __( 'Import Courses', 'lifterlms' ),
+				'skip'  => __( 'Skip this step', 'lifterlms' ),
 			),
-			admin_url()
 		);
-	}
 
-	/**
-	 * Get an array of step slugs => titles
-	 *
-	 * @since 3.0.0
-	 *
-	 * @return array
-	 */
-	public function get_steps() {
-
-		$steps = array(
-			'intro'    => __( 'Welcome!', 'lifterlms' ),
-			'pages'    => __( 'Page Setup', 'lifterlms' ),
-			'payments' => __( 'Payments', 'lifterlms' ),
-			'coupon'   => __( 'Coupon', 'lifterlms' ),
-			'finish'   => __( 'Finish!', 'lifterlms' ),
-		);
+		$this->add_hooks();
+
+		// Add HTML around importable courses on last step.
+		add_action( 'llms_before_importable_course', array( $this, 'output_before_importable_course' ) );
+		add_action( 'llms_after_importable_course', array( $this, 'output_after_importable_course' ) );
 
-		/**
-		 * Filter the steps included in the setup wizard
-		 *
-		 * @since 4.8.0
-		 *
-		 * @param string[] $steps Array of setup wizard steps. The array key is the slug/id of the step and the array value
-		 *                        is the step's title displayed in the wizard's navigation.
-		 */
-		return apply_filters( 'llms_setup_wizard_steps', $steps );
+		// Hide action buttons on importable courses during last step.
+		add_filter( 'llms_importable_course_show_action', '__return_false' );
```

Comment:
The name of this hook is too much generic to me, I would go with the `llms_setup_wizard_` prefix...
What do you think?

Reply:
@eri-trabiccolo Do you know if the importable courses are used outside of the wizard or if any add-ons would already be making use of these filters?

Reply:
@seothemes 
don't consider this, my bad `'llms_importable_course_show_action'` is defined elsewhere you don't need to change anything here.